paper-dialog.html

```diff
@@ -70,40 +61,206 @@ <h2>Header</h2>
+        }
+      }
+
+      @keyframes keyFrameScaleDown {
+        0% {
+          transform: scale(1.0);
+        }
+        100% {
+          transform: scale(0.0);
+        }
+      }
+
+      :host(.fade-in-animation) {
+        opacity: 0;
+        animation-delay: 0;
+        animation-name: keyFrameFadeInOpacity;
+        animation-iteration-count: 1;
+        animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+        animation-duration: var(--paper-dialog-duration-in, 500ms);
+        animation-fill-mode: forwards;
+        @apply --paper-dialog-animation;
+      }
+
+      :host(.fade-out-animation) {
+        opacity: 1;
+        animation-delay: 0;
+        animation-name: keyFrameFadeOutOpacity;
+        animation-iteration-count: 1;
+        animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+        animation-duration: var(--paper-dialog-duration-out, 1500ms);
+        animation-fill-mode: forwards;
+        @apply --paper-dialog-animation;
+      }
+
+      :host(.scale-up-animation) {
+        transform: scale(0);
+        opacity: 1;
+        animation-delay: 0;
+        animation-name: keyFrameScaleUp;
+        animation-iteration-count: 1;
+        animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+        animation-duration: var(--paper-dialog-duration-in, 500ms);
+        animation-fill-mode: forwards;
+        @apply --paper-dialog-animation;
+      }
+
+      .scale-down-animation {
+        transform: scale(1);
+        opacity: var(--paper-dialog-opacity, 0.9);
+        animation-delay: 0;
+        animation-name: keyFrameScaleDown;
+        animation-iteration-count: 1;
+        animation-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+        animation-duration: var(--paper-dialog-duration-out, 500ms);
+        animation-fill-mode: forwards;
+        @apply --paper-dialog-animation;
+      }      
+
+    </style>
     <slot></slot>
   </template>
 </dom-module>
 
 <script>
-Polymer({
-  is: 'paper-dialog',
-
-  behaviors: [
-    Polymer.PaperDialogBehavior,
-    Polymer.NeonAnimationRunnerBehavior
-  ],
-
-  listeners: {
-    'neon-animation-finish': '_onNeonAnimationFinish'
-  },
-
-  _renderOpened: function() {
-    this.cancelAnimation();
-    this.playAnimation('entry');
-  },
-
-  _renderClosed: function() {
-    this.cancelAnimation();
-    this.playAnimation('exit');
-  },
-
-  _onNeonAnimationFinish: function() {
-    if (this.opened) {
-      this._finishRenderOpened();
-    } else {
-      this._finishRenderClosed();
+  Polymer({
+    is: 'paper-dialog',
+
+    behaviors: [
+      Polymer.PaperDialogBehavior
+    ],
+
+    listeners: {
+      'webkitAnimationEnd': '_onAnimationEnd',
+    },
+
+    properties: {
+      /**
+       * Set Entry Animation 
+       */
+      entryAnimation: {
+        value: "fade-in-animation",
+        type: String
+      },
+      /**
+       * Set Exit Animation
+       */
+      exitAnimation: {
+        type: String
+      },
+      _showing: {
```

Comment:
this should not be required, as it's enough to use `this.opened` (defined in `IronOverlayBehavior` which is inherited by `PaperDialogBehavior`)